Adds an output to docs files

sarahsimpers opened the pull request:
## Proposed changes

This experimental PR adds an output section to the docs files. It would depend on an "output" annotation, which would use the template already defined in each file (listTemplate, describeTemplate, etc) that reflects the output in the CLI itself.

I tested this locally and I am including screenshots of create, delete, list, and describe commands after a local test with these changes. I also included a screenshot of what the annotation looks like.

<img width="676" alt="Screenshot 2023-03-12 at 3 12 19 PM" src="https://user-images.githubusercontent.com/82042374/224567878-5c2e8476-a33d-4ba8-8b35-9356bd4d9a54.png">
<img width="683" alt="Screenshot 2023-03-12 at 3 13 00 PM" src="https://user-images.githubusercontent.com/82042374/224567882-9dfec595-6dc4-49e0-9ce7-19135a837830.png">
<img width="678" alt="Screenshot 2023-03-12 at 3 13 23 PM" src="https://user-images.githubusercontent.com/82042374/224567898-e068f459-228b-4a7d-ad39-55598c1261c3.png">
<img width="676" alt="Screenshot 2023-03-12 at 3 14 45 PM" src="https://user-images.githubusercontent.com/82042374/224567904-0df90ffe-38e4-4433-a7eb-a76b1a442fdb.png">
<img width="493" alt="Screenshot 2023-03-12 at 3 23 31 PM" src="https://user-images.githubusercontent.com/82042374/224568025-57da2e63-52d7-4503-81bf-e159ee1236dd.png">

_Jira ticket:_ 

N/A

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

sarahsimpers commented:
@gssbzn, could you take a look at this PR when you have time and let me know if the Go code looks ok?
@melissamahoney-mongodb, let me know what you think about making this change to add an output to the docs/whether you have any copy suggestions. It won't take long to add the required annotations, I added them for all create commands while testing.

Thank you both!!

gssbzn commented on output.go:
```diff
@@ -0,0 +1,64 @@
+	w := new(tabwriter.Writer)
+	w.Init(buf, 0, 0, 0, ' ', tabwriter.Debug|tabwriter.AlignRight)
+
+	if cmd.Annotations["output"] == "" {
+		return
+	}
```
you probably want to do this first to save some cpu cycles

gssbzn commented on output.go:
```diff
@@ -0,0 +1,64 @@
+// This function can return the output for all commands when the output template is added as an annotation in the command file
+
+func printOutputCreate(buf *bytes.Buffer, cmd *cobra.Command) {
+	removerange := strings.ReplaceAll(cmd.Annotations["output"], "{{range .Results}}", "")
+	removeend := strings.ReplaceAll(removerange, "{{end}}", "")
+	bracketsremoved1 := strings.ReplaceAll(removeend, "{{.", "<")
+	bracketsremoved2 := strings.ReplaceAll(bracketsremoved1, "}}", ">")
+	replacevariable := strings.ReplaceAll(bracketsremoved2, "%s", "<Name>")
+	replacevariable2 := strings.ReplaceAll(replacevariable, "\n", "\n   ")
```
[q] can we use the same string? like
```suggestion
	output := strings.ReplaceAll(cmd.Annotations["output"], "{{range .Results}}", "")
	output = strings.ReplaceAll(output, "{{end}}", "")
	output = strings.ReplaceAll(output, "{{.", "<")
	output = strings.ReplaceAll(output, "}}", ">")
	output = strings.ReplaceAll(output, "%s", "<Name>")
	output = strings.ReplaceAll(output, "\n", "\n   ")
```

also not sure if this is the best option from the top of my head to do multi string replacement on go

gssbzn commented on output.go:
```diff
@@ -0,0 +1,64 @@
+	bracketsremoved2 := strings.ReplaceAll(bracketsremoved1, "}}", ">")
+	replacevariable := strings.ReplaceAll(bracketsremoved2, "%s", "<Name>")
+	replacevariable2 := strings.ReplaceAll(replacevariable, "\n", "\n   ")
+	w := new(tabwriter.Writer)
+	w.Init(buf, 0, 0, 0, ' ', tabwriter.Debug|tabwriter.AlignRight)
```
not sure if this matters but these are the values we use for the tabwritter on the cli
https://github.com/mongodb/mongodb-atlas-cli/blob/7a304c4870c026d80e5b26391efad0e2c48f0f26/internal/templatewriter/templatewriter.go#L23-L34

sarahsimpers replied:
I experimented with these a bit and made some changes to the code overall to reflect a different alignment and padding for this. The thing that has me stumped right now: when I add 3 spaces to the beginning of the first line for these, it messes up the alignment of all other headers by 3 spaces. But without the 3 spaces it won't be in the code box.
**With 3 extra spaces at the beginning: ID aligns correctly, others are 3 spaces too far in:**
![Screenshot 2023-03-14 at 9 27 46 AM](https://user-images.githubusercontent.com/82042374/225016101-98fcfc57-2ac4-4d5f-9f20-ffe1fabe4e10.png)
**Without 3 extra spaces at the beginning: All except ID align correctly:**
![Screenshot 2023-03-14 at 9 31 34 AM](https://user-images.githubusercontent.com/82042374/225016738-f717f6e1-53dc-41ff-bf90-86900a3d76e0.png)

gssbzn commented on output.go:
```diff
@@ -0,0 +1,64 @@
+.. code-block::
+
+`)
+	if strings.HasSuffix(cmd.Annotations["output"], "}"+"\n") {
```
```suggestion
	if strings.HasSuffix(cmd.Annotations["output"], "}\n") {
```

gssbzn reviewed (commented):
general lgtm with some minor go recommendations, this looks like a cool addition

melissamahoney-mongodb commented on output.go:
```diff
@@ -0,0 +1,65 @@
+
+	buf.WriteString(outputHeader)
+	buf.WriteString(`
+If the command succeeds, the CLI prints a message similar to the following and replaces the values in brackets with your values:
```
Nit: I know this is definitely a nit.. For some reason to me "prints a message" sounds like the returns are informational instead of possible data/actions. The CLI also isn't really doing a replace, we're just trying to represent placeholders. I've made a suggestion but I'm not sure its that much better, up to you.

```suggestion
If the command succeeds, the CLI returns output similar to the following sample. Values in brackets represent your values.
```

sarahsimpers commented:
@gssbzn @melissamahoney-mongodb Ready for another look! I made both of your suggested changes and I fixed the formatting on the responses with columns. Updated screenshots of the results are in the PR description. Thanks!

gssbzn reviewed (approved):
Amazing addition here
